district_data_import.py

```diff
@@ -202,6 +202,28 @@
         if dp_donorrecord_for_update['SPOUSE_EMAIL']:
             dp_donorrecord['SPOUSE_EMAIL'] = dp_donorrecord_for_update['SPOUSE_EMAIL']
 
+        #after updating emails, deail with mailmerge fname fields
+        #we always want to have a real first name in mailmerge fname if there is email for the donor,
+        #particularly after getting any updates from district and after swapping parents.
+        if dp_donorrecord['EMAIL']:
+            mailmerge_fname = dp_donorrecord['MAILMERGE_FNAME']
+            if not mailmerge_fname or mailmerge_fname.lower() == 'no email':
+                dp_donorrecord['MAILMERGE_FNAME'] = dp_donorrecord['FIRST_NAME']
+        else:
+            dp_donorrecord['MAILMERGE_FNAME'] = 'no email'
+
+        #for the spouse mailmerge, it can be 'no email' or set to first name
+        #set it to first_name (or keep the existing name) if they have different emails.
+        #set to no email if they have the same emails or do not have spouse email.
+        if dp_donorrecord['SPOUSE_EMAIL']:
+            if dp_donorrecord['EMAIL'].lower() == dp_donorrecord['SPOUSE_EMAIL'].lower():
+                dp_donorrecord['SP_MAILMERGE_FNAME'] = 'no email'
+            elif dp_donorrecord['SP_MAILMERGE_FNAME'].lower() == 'no email':
```

Comment:
Can you create a local variable here like you do for `MAILMERGE_FNAME`? For that field you check against null but you don't do that in this case.

Reply:
I can create a local variable for sp_mailmerge_fname.
I don't check against null for sp_mailmerge_fname because I don't have to.  
The first if (email == spouse_email) checks to see if it's different or the same email.
If it's same email the spouse get 'no email'.
otherwise (this means they have different email) if spouse is set to 'no email' then set it to first name of the spouse.
otherwise we're keeping whatever is already there (this would be the check for not null that's not needed).

Reply:
I initially didn't create a variable for sp_mailmerge_fname because I would only use it once (on line 221) and I didn't see a need.  I can create one if you still think it's needed.

Reply:
ok, I've re-read my reasoning and I'm wrong   I would need to check for null there, too.  So I'll re-write it.